File: Lab6/t2.cpp
#include <stdio.h>
int judgePrime(int a);
int main(){
    int num;
    do{
        printf("Please input a positive integer (input 0 to exit): ");
        scanf("%d", &num);
        if (num < 0){
            puts("Invalid input!");
            continue;
        }
        else if (num == 0){
            puts("Bye-bye!");
            break;
        }
        if (judgePrime(num))
            printf("%d is a prime.\n", num);
        else
        printf("%d is not a prime.\n", num);
    }while(true);
    return 0;
}
int judgePrime(int a){
    int i;
    if (a == 1)
        return 0;
    for (i = 2; i < a; i++){
        if (a % i == 0)
            return 0;
    }
    return 1;
}

File: Lab6/t2.java
import java.util.Scanner;
public class t2{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        int num;
        do {
            System.out.print("Please input a positive integer (input 0 to exit): ");
            num = scan.nextInt();
            if (num < 0){
                System.out.println("Invalid Input!");
                continue;
            }
            else if (num == 0){
                System.out.println("Bye-bye");
                break;
            }
            if (judgePrime(num))
                System.out.printf("%d is a prime.\n", num);
            else
                System.out.printf("%d is not a prime.\n", num);
        }while (true);
        scan.close();
    }
    private static boolean judgePrime(int number){
        if (number == 1)
            return false;
        for (int i = 2; i <= Math.sqrt(number); i++){
            if (number % i == 0)
                return false;
        }
        return true;
    }
}